components/KiboOrderSummary.vue

```diff
@@ -0,0 +1,242 @@
+        />
+      </slot>
+    </div>
+    <div class="highlighted sf-order-summary__promo-code">
+      <slot name="promo" v-bind="{ promoCode }">
+        <SfInput
+          v-model="promoCode"
+          name="promoCode"
+          label="Enter promo code"
+          class="sf-input--filled sf-order-summary__promo-code-input"
+        />
+        <SfButton
+          class="sf-order-summary__promo-code-button"
+          data-testid="apply-button"
+          @click="$emit('click:apply-code')"
+        >
+          Apply
+        </SfButton>
+      </slot>
+    </div>
+    <div class="sf-order-summary__characteristics">
+      <slot name="characteristics" v-bind="{ characteristics }">
+        <SfCharacteristic
+          v-for="characteristic in characteristics"
+          :key="characteristic.title"
+          :title="characteristic.title"
+          :description="characteristic.description"
+          :icon="characteristic.icon"
+          size-icon="32px"
+          color-icon="green-primary"
+          class="sf-order-summary__characteristics-item"
+        >
+        </SfCharacteristic>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import {
+  SfHeading,
+  SfButton,
+  SfDivider,
+  SfProperty,
+  SfCharacteristic,
+  SfInput,
+} from "@storefront-ui/vue"
+export default {
+  name: "SfOrderSummary",
+  components: {
+    SfHeading,
+    SfButton,
+    SfDivider,
+    SfProperty,
+    SfCharacteristic,
+    SfInput,
+  },
+  props: {
+    orderTitle: {
+      type: String,
+      default: "Order review",
+    },
+    orderTitleLevel: {
+      type: Number,
+      default: 2,
+    },
+    order: {
+      type: Object,
+      default: () => ({}),
+    },
+    propertiesNames: {
+      type: Array,
+      default: () => ["Products", "Subtotal", "Shipping", "Total price"],
+    },
+    characteristics: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  setup(props) {
+    const { order } = props
+
+    const products = computed(() => {
+      return order.orderItems
+    })
+
+    const totalItems = computed(() => {
+      return products.value
+        ? products.value.reduce((previous, current) => {
+            return previous + current.qty
+          }, 0)
+        : 0
+    })
+
+    const shipping = computed(() => {
+      return order.shipping || {}
+    })
+
+    const shippingMethod = computed(() => {
+      return shipping.value.shippingMethod || { price: "" }
+    })
+
+    const subtotal = computed(() => {
+      const subtotal = products.value
+        ? products.value.reduce((previous, current) => {
+            const qty = current.qty
+            const price = current.price.special ? current.price.special : current.price.regular
+            const total = qty * parseFloat(price?.replace("$", ""))
+            return previous + total
+          }, 0)
+        : 0
+      return "$" + subtotal.toFixed(2)
+    })
+
+    const total = computed(() => {
```

Comment:
We need to create getters for all of these values, we never want to calculate ourselves